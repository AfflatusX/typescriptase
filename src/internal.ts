export abstract class Renderable {
  public abstract render(ctx: IRenderContext): string;
}

export abstract class Composable extends Renderable {}

export interface IRenderContext {
  name: string;
  path: string;
}
