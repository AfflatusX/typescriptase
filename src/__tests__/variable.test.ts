import { BasicVariable, ComplexVariable } from "./internal";

test(
  "BasicVariable",
  async (): Promise<void> => {
    const output: BasicVariable = new BasicVariable();
    expect(output.render()).toMatchSnapshot();
  },
);

test(
  "ComplexVariable",
  async (): Promise<void> => {
    const output: ComplexVariable = new ComplexVariable();
    expect(output.render()).toMatchSnapshot();
  },
);
