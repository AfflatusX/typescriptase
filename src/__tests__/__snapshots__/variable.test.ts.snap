// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicVariable 1`] = `"const basicTest: string;"`;

exports[`ComplexVariable 1`] = `"export let complexTest: string | null = \\"complex test\\";"`;
